File: project/core/module/rest-service/src/main/java/vdi/service/rest/conversion/BrokenImportDetails.java
package vdi.service.rest.conversion;

import vdi.core.db.cache.model.BrokenImportRecord;
import vdi.service.rest.generated.model.BrokenImportDetailsImpl;

public class BrokenImportDetails extends BrokenImportDetailsImpl {
  public BrokenImportDetails(BrokenImportRecord record) {
    setDatasetId(record.getDatasetIdAsString());
    setOwner(record.getOwnerID().toLong());
    setDatasetType(new DatasetTypeOutput(record.getType()));
    setInstallTargets(record.getInstallTargets());
    setMessages(record.getMessages());
  }
}

File: project/core/module/rest-service/src/main/kotlin/vdi/service/rest/server/outputs/BrokenImportDetails.kt
package vdi.service.rest.server.outputs

import vdi.core.db.cache.model.BrokenImportRecord
import vdi.service.rest.generated.model.BrokenImportDetails
import vdi.service.rest.generated.model.BrokenImportDetailsImpl

fun BrokenImportDetails(record: BrokenImportRecord): BrokenImportDetails =
  BrokenImportDetailsImpl().apply {
    datasetId = record.datasetID.toString()
    owner = record.ownerID.toLong()
    datasetType = record.type.toExternal()
    installTargets = record.projects
    messages = record.messages
  }
